src/cascadia/TerminalApp/CommandPalette.h

```diff
@@ -13,21 +13,33 @@ namespace winrt::TerminalApp::implementation
         CommandPalette();
 
         Windows::Foundation::Collections::IObservableVector<TerminalApp::Command> FilteredActions();
-        void SetActions(Windows::Foundation::Collections::IVector<TerminalApp::Command> const& actions);
+
+        void SetCommandPaletteActions(Windows::Foundation::Collections::IVector<TerminalApp::Command> const& actions);
+        void EnableCommandPaletteMode();
 
         void SetDispatch(const winrt::TerminalApp::ShortcutActionDispatch& dispatch);
 
+        // TabSwitcherMode Specific
+        void EnableTabSwitcherMode(const Windows::System::VirtualKey& anchorKey);
+        void OnTabsChanged(const Windows::Foundation::IInspectable& s, const Windows::Foundation::Collections::IVectorChangedEventArgs& e);
+
     private:
         friend struct CommandPaletteT<CommandPalette>; // for Xaml to bind events
 
         Windows::Foundation::Collections::IObservableVector<TerminalApp::Command> _filteredActions{ nullptr };
         Windows::Foundation::Collections::IVector<TerminalApp::Command> _allActions{ nullptr };
+
+        Windows::Foundation::Collections::IVector<TerminalApp::Command> _allCommandPaletteActions{ nullptr };
```

Comment:
You know what, I think we're trying to solve similar problems both in this PR and in #6856. In that PR, I added `CommandPalette::_commandsToFilter()`, to get the list of Commands to filter. If this merges first, I'm fine dealing with the merge conflict to handle that.

Reply:
I actually like the way you're going about it in your PR, I'll copy it over here! I've actually got a lot more polish to do here so I think that PR is gonna merge first, I don't mind dealing with the conflict (shouldn't be that much if I copy everything including method names etc.)